src/ssh.rs

```diff
@@ -1,12 +1,21 @@
 use failure::Error;
+use failure::Fail;
 use failure::ResultExt;
 use slog;
 use ssh2;
+use std::fs::File;
+use std::io;
 use std::net::{SocketAddr, TcpStream};
 use std::path::Path;
 use std::thread;
 use std::time::{Duration, Instant};
 
+#[derive(Debug, Fail)]
+enum SshError {
```

Comment:
This can probably be a `struct FileTransferError` rather than an `enum`, since there aren't any other variants :)

Reply:
Done, fair :)